test/commands/force/data/soql/query/dataSoqlQuery.nut.ts

```diff
@@ -59,30 +60,37 @@ function runQuery(
 
 describe('data:soql:query command', () => {
   let testSession: TestSession;
-  let hubOrgUsername: string;
+  let hubOrgUsername: string | undefined;
 
   before(async () => {
     testSession = await TestSession.create({
-      setupCommands: [
-        'sfdx force:org:create -f config/project-scratch-def.json --setdefaultusername --wait 10 --durationdays 1',
-        'sfdx force:source:push',
-        'sfdx config:get defaultdevhubusername --json',
+      scratchOrgs: [
+        {
+          executable: 'sfdx',
+          config: 'config/project-scratch-def.json',
+          setDefault: true,
+        },
       ],
       project: { sourceDir: path.join('test', 'test-files', 'data-project') },
+      devhubAuthStrategy: 'AUTO',
     });
+    // get default devhub username
+    hubOrgUsername = execCmd<[{ key: string; value: string }]>('config:get defaultdevhubusername --json', {
```

Comment:
you can access the devhub this way, `testSession.hubOrg`

https://github.com/salesforcecli/cli-plugins-testkit/blob/main/src/testSession.ts#L109